### PR TITLE
[INLONG-11035][SDK] Transform support TRUNCATE() function

### LINKED ISSUE #11035: [Feature][SDK] Transform support TRUNCATE(numeric1, integer2) function
### Description

* parent issue: #10796 
* from flink sql: https://nightlies.apache.org/flink/flink-docs-release-1.17/zh/docs/dev/table/functions/systemfunctions/#%E7%AE%97%E6%9C%AF%E5%87%BD%E6%95%B0
* TRUNCATE(numeric1, integer2)--returns the number that intercepts integer2 decimal places. If numeric1 or integer2 is NULL, NULL is returned. If integer2 is 0, the result has no decimal point or fractional part. integer2 can be negative, making the integer2 digit to the left of the decimal point of the value zero. This function can also be used by passing only one numeric1 argument without setting Integer2. If Integer2 is not set, Integer2 is 0. 
* For example: 
truncate(42.324, 2)--return 42.32 
truncate(42.324)--return 42.0

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by emptyOVO**
Fixes #11035 

### Motivation
add TruncateFunction class and provide unit test

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private ValueParser bigDecimalParser;
+
+    private ValueParser integerParser;
+
+    public TruncateFunction(Function expr) {
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        if (expressions != null) {
+            bigDecimalParser = OperatorTools.buildParser(expressions.get(0));
+            if (expressions.size() == 2) {
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java:51]: It is acceptable for "expressions.size() > 2"

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java]: done

**Review by aloyszhang — APPROVED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+/**
+ * TruncateFunction
+ * description: returns the number that intercepts integer2 decimal places.
```
Comment (vernedeng) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java:33]: please add the example of this function

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TruncateFunction.java:33]: done

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**